File: 백준/Silver/2960. 에라토스테네스의 체/에라토스테네스의 체.py
n, k = map(int, input().split())

arr = [0] * (n+1)
cnt = 0
for i in range(2, n+1):
  for j in range(i, n+1, i):
    if arr[j] == 0:
      arr[j] = 1
      cnt += 1
      if cnt == k:
        print(j)
        exit()

File: 백준/Silver/2960. 에라토스테네스의 체/에라토스테네스의 체.c
#include<stdio.h>

int main(void){
  int n, k, cnt = 0;

  scanf("%d %d", &n, &k);
  int arr[n];
  for (int i = 0; i < n+1; i++) arr[i] = 0;

  for(int i = 2; i < n+1; i++){
    for(int j = i; j < n+1; j+=i){
      if(arr[j] == 0){
        arr[j] = 1;
        cnt ++;
        if(cnt == k){
          printf("%d\n", j);
          return 0;
        }
      }
    }
  }
}